### PR TITLE
move port splitting to common place; add to node resource location

### PR DESCRIPTION
**Pull request by lavalamp**

### CODE REVIEW

**Comment by lavalamp** [comment]
To unblock the UI PR.

**Comment by preillyme** [comment]
@lavalamp: I'll try now.

**Comment by preillyme** [comment]
@lavalamp this seems to work as expected. I can now do requests like: ```https://104.154.67.146/api/v1beta2/proxy/nodes/kubernetes-minion-grly.c.shared-kraken.internal:4194/api/v1.1/machine/``` thanks!

File: pkg/util/port_split.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// Takes a string of the form "name:port" or "name".
+//  * If id is of the form "name" or "name:", then return (name, "", true)
+//  * If id is of the form "name:port", then return (name, port, true)
+//  * Otherwise, return ("", "", false)
+// Additionally, name must be non-empty or valid will be returned false.
+//
+// Port is returned as a string, and it is not required to be numeric (could be
+// used for a named port, for example).
+func SplitPort(id string) (name, port string, valid bool) {
```
Comment (brendandburns) [inline pkg/util/port_split.go]: given that in all cases you test valid and then return an error, perhaps just return an error instead of the 'valid' bool?

**Comment by brendandburns** [comment]
Basically LGTM, small comment on a refactor that I think makes the code cleaner.

File: pkg/util/port_split.go (hunk 1 above)
Comment (lavalamp) [inline pkg/util/port_split.go]: The error is different in all three places, though... I also don't think util can return an api error (dependency goes the other direction), and if I return a regular error, then it has to get wrapped in all three places anyway.

**Comment by preillyme** [comment]
@brendanburns can this be merged as-is as I think @lavalamp is correct unless I'm missing something.

**Comment by piosz** [comment]
I agree with @lavalamp.

File: pkg/util/port_split.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+limitations under the License.
+*/
+
+package util
+
+import (
+	"strings"
+)
+
+// Takes a string of the form "name:port" or "name".
```
Comment (thockin) [inline pkg/util/port_split.go]: This is very similar to but different from http://golang.org/pkg/net/#SplitHostPort - is it worthwhile, really?  or should we thunk down to that when we can?

Comment (lavalamp) [inline pkg/util/port_split.go]: Yeah, we can investigate whether we can just use that. I wanted to at least get these three cases consistent, though.

**Pull request merged**